spring-web/src/main/java/org/springframework/http/converter/HttpMessageConversionException.java

```diff
@@ -17,22 +17,29 @@
 package org.springframework.http.converter;
 
 import org.springframework.core.NestedRuntimeException;
+import org.springframework.http.HttpStatus;
+
+import java.util.Optional;
 
 /**
  * Thrown by {@link HttpMessageConverter} implementations when a conversion attempt fails.
  *
  * @author Arjen Poutsma
+ * @author Sebastien Deleuze
  * @since 3.0
  */
 @SuppressWarnings("serial")
 public class HttpMessageConversionException extends NestedRuntimeException {
 
+	private final Optional<HttpStatus> errorStatus;
```

Comment:
We should generally remain defensive against overuse of `Optional`. So agreed, let's avoid it as field type and only expose it for method signatures where it nicely aligns with typical use of the method.